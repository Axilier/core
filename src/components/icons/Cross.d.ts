import { IconProps } from '../../Types';
export declare const Cross: ({ style, className, iconColor, }: IconProps) => JSX.Element;
