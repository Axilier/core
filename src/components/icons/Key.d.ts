import { IconProps } from '../../Types';
export declare const Key: ({ iconColor, style, className, }: IconProps) => JSX.Element;
