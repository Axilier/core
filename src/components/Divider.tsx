import React from 'react';
import { DividerProps } from '../Types';
import Layout from './Layout';

const Divider = ({ children, style }: DividerProps): JSX.Element => (
    <Layout
        orientation={'row'}
        style={{
            ...style,
            alignItems: 'center',
            width: '100%',
            fontFamily: "'IBM Plex Sans', sans-serif",
        }}
    >
        <div
            style={{
                backgroundColor: '#BEBEBE',
                width: '100%',
                height: '2px',
            }}
        />
        <div
            style={{
                padding: '0 8px',
                color: '#BEBEBE',
                verticalAlign: 'center',
            }}
        >
            {children}
        </div>
        <div
            style={{
                backgroundColor: '#BEBEBE',
                width: '100%',
                height: '2px',
            }}
        />
    </Layout>
);

export default Divider;
