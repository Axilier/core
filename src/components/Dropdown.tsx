/** @format */

// /** @format */
//
// // @flow
// import * as React from 'react';
// import { useEffect, useState } from 'react';
// import { DropdownProps } from '../Types';
//
// const Dropdown = ({ open, onChange, style, className }: DropdownProps) => {
//     const [isOpen, setIsOpen] = useState(open);
//
//     useEffect(() => {
//         setIsOpen(open);
//     }, [open]);
//
//     return (
//         <svg
//             transform={isOpen ? 'rotate(90)' : 'none'}
//             className={`drop-down-icon ${className}`}
//             onClick={() => {
//                 if (!onChange) return;
//                 onChange();
//             }}
//             viewBox={'0 0 64 64'}
//             xmlns={'http://www.w3.org/2000/svg'}
//             fillRule={'evenodd'}
//             clipRule={'evenodd'}
//             strokeLinejoin={'round'}
//             strokeMiterlimit={2}
//             style={style}
//         >
//             <g transform={'matrix(2.90892 0 0 2.9091 -4174.32 -1000.725)'}>
//                 <circle cx={1446} cy={355} r={11} fill={'#ebebeb'} />
//                 <path
//                     d={
//                         'M1446 344c6.07 0 11 4.929 11 11s-4.93 11-11 11-11-4.929-11-11 4.93-11 11-11zm0 .174c5.98 0 10.83 4.851 10.83 10.826 0 5.975-4.85 10.826-10.83 10.826-5.97 0-10.83-4.851-10.83-10.826 0-5.975 4.86-10.826 10.83-10.826z'
//                     }
//                     fill={'#d3d3d3'}
//                 />
//             </g>
//             <path
//                 d={'M47.297 31.109L23.8 47.997V14.22l23.497 16.889z'}
//                 fill={'#d3d3d3'}
//             />
//         </svg>
//     );
// };
//
// export default Dropdown;
