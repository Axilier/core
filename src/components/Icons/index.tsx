/** @format */

export { Lock } from './Lock';
export { Room } from './Room';
export { Mail } from './Mail';
export { Key } from './Key';
export { Google } from './Google';
export { Github } from './Github';
export { Warning } from './Warning';
export { Import } from './Import';
export { Tick } from './Tick';
export { Cross } from './Cross';
export { Question } from './Question';
export { Issue } from './Issue';
export { Copyright } from './Copyright';
export { Connections } from './Connections';
export { SharedFiles } from './SharedFiles';
export { Favourites } from './Favourites';
export { Recent } from './Recent';
export { AllFiles } from './AllFiles';
export { Search } from './Search';
export { Dropdown } from './Dropdown';
