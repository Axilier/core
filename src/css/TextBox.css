.core-text-box {
    justify-content: flex-start;
    margin: 10px;
    font-family: 'IBM Plex Sans', 'Source Sans Pro', sans-serif;
    display: flex;
    font-size: 13px;
    flex-wrap: wrap;
    text-transform: capitalize;

}

.core-text-box-label {
    display: flex;
    flex-direction: row;
    margin-right: 5px;
    font-size: 13px;
}

.core-text-box-input-units {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 0 10px;
}

.core-text-box-input {
    padding-top: 0;
    padding-bottom: 0;
    align-items: center;
    border: none;
    font-size: 13px;
    margin-left: 10px;
}

.core-text-box-input:focus {
    outline: none;
}

.core-required-icon {
    color: #FF0000;
    margin-left: 3px;
}
