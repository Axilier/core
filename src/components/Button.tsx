/** @format */

// @flow
import * as React from 'react';
import { useEffect, useState } from 'react';
import { ButtonProps } from '../Types';
import '../css/Button.css';

const Button = ({
    label,
    type,
    variant,
    onClick,
    style,
    className,
    disabled,
    buttonColor,
    altButtonColor,
    buttonIcon,
    iconBackgroundColor,
    size,
}: ButtonProps) => {
    const [clicked, setClicked] = useState(false);
    const [hover, setHover] = useState(false);

    useEffect(() => console.log(hover), [hover]);

    const primaryColour = (): string => {
        switch (type) {
            case 'primary':
                return buttonColor || '#057AFF';
            case 'secondary':
                return buttonColor || 'white';
            case 'tertiary':
                return buttonColor || 'black';
            default:
                return buttonColor || 'black';
        }
    };
    const secondaryColour = (): string => {
        switch (type) {
            case 'primary':
                return altButtonColor || 'white';
            case 'secondary':
                return altButtonColor || '#057AFF';
            case 'tertiary':
                return altButtonColor || 'white';
            default:
                return altButtonColor || 'white';
        }
    };

    const border = () => {
        switch (variant) {
            case 'contained':
                return hover
                    ? `solid 2px ${primaryColour()}`
                    : 'solid 2px transparent';
            case 'outlined':
                return `solid 2px ${primaryColour()}`;
            case 'text':
                return 'none';
            default:
                return 'none';
        }
    };
    const backgroundColor = () => {
        switch (variant) {
            case 'contained':
                return hover ? secondaryColour() : primaryColour();
            case 'outlined':
                return hover ? primaryColour() : 'transparent';
            case 'text':
                return 'transparent';
            default:
                return primaryColour();
        }
    };
    const color = () => {
        switch (variant) {
            case 'contained':
                return hover ? primaryColour() : secondaryColour();
            case 'outlined':
                return hover ? secondaryColour() : primaryColour();
            case 'text':
                return primaryColour();
            default:
                return primaryColour();
        }
    };

    const btnSize = (): string => {
        switch (size) {
            case 'small':
                return '150px';
            case 'large':
                return '320px';
            default:
                return size || '';
        }
    };

    return (
        <div
            className={'button-container'}
            style={{ width: variant === 'text' ? '' : btnSize() }}
        >
            <button
                onMouseDown={() => setClicked(true)}
                onMouseUp={() => setClicked(false)}
                onMouseEnter={() => setHover(true)}
                onMouseLeave={() => setHover(false)}
                type={'button'}
                className={`button ${className}`}
                style={{
                    border: border(),
                    backgroundColor: backgroundColor(),
                    color: color(),
                    padding: variant === 'text' ? 'none' : '5px 35px',
                    width: variant === 'text' ? '' : btnSize(),
                    ...style,
                }}
                onClick={() => {
                    if (!onClick) return;
                    onClick();
                }}
            >
                {buttonIcon ? (
                    <div
                        style={{
                            backgroundColor: iconBackgroundColor || 'white',
                        }}
                    >
                        {buttonIcon}
                    </div>
                ) : null}
                {label}
            </button>
            {disabled ? (
                <div
                    className={'button-filter'}
                    style={{
                        opacity: 0.3,
                    }}
                />
            ) : null}
        </div>
    );
};

Button.defaultProps = {
    variant: 'outlined',
    type: 'primary',
    label: 'test',
    onClick: () => {},
    disabled: false,
    size: 'small',
};

export default Button;
