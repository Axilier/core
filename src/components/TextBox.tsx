/** @format */

// @flow
import * as React from 'react';
import { ChangeEvent, useState } from 'react';
import { TextBoxProps } from '../Types';
import '../css/TextBox.css';

const TextBox = ({
    label,
    prefixComponent,
    suffixComponent,
    filter,
    placeholder,
    size,
    variant,
    required,
    disabled,
    maxLength,
    units,
    onChange,
}: TextBoxProps) => {
    const [value, setValue] = useState('');

    function handleChange(evt: ChangeEvent<HTMLInputElement>) {
        const isFiltered = filter ? filter(evt.target.value) : true;
        if (!isFiltered) return;
        setValue(evt.target.value);
        if (!onChange) return;
        onChange(evt.target.value);
    }

    const iconStyles = (component: JSX.Element | undefined) => ({
        display: !component ? 'none' : 'block',
        height: size === 'small' ? '10x' : '16px',
        width: size === 'small' ? '12x' : '20px',
        marginLeft: '10px',
    });

    return (
        <div
            className={'text-box'}
            style={{
                flexDirection: variant === 'filled' ? 'row' : 'column',
                alignItems: variant === 'filled' ? 'center' : 'start',
            }}
        >
            {label !== '' ? (
                <div className={'text-box-label'}>
                    {label}
                    {required ? <div className={'required-icon'}>*</div> : null}
                </div>
            ) : null}
            <div
                className={'text-box-input-units'}
                style={{
                    color: disabled ? '#8C8C8C' : '#000000',
                    height: size === 'small' ? '34px' : '44px',
                    borderRadius: variant === 'outlined' ? '5px' : '0',
                    border:
                        variant === 'outlined' ? 'solid 2px #057EFF' : 'none',
                }}
            >
                <div style={iconStyles(prefixComponent)}>{prefixComponent}</div>
                <input
                    value={value}
                    className={'text-box-input'}
                    placeholder={placeholder}
                    style={{
                        width: size === 'small' ? '30px' : '320px',
                        height: size === 'small' ? '34px' : '44px',
                        cursor: disabled ? 'not-allowed' : 'text',
                        backgroundColor:
                            variant === 'filled' ? '#F3F3F3' : 'transparent',
                    }}
                    onChange={(evt) => handleChange(evt)}
                    maxLength={maxLength || -1}
                    disabled={disabled}
                />
                <div style={iconStyles(suffixComponent)}>{suffixComponent}</div>
                <div
                    style={{
                        display: !units ? 'none' : 'block',
                        marginLeft: '5px',
                    }}
                >
                    {units}
                </div>
            </div>
        </div>
    );
};

TextBox.defaultProps = {
    units: '',
    disabled: false,
    label: '',
    size: 'small',
    variant: 'filled',
    onChange: () => null,
};

export default TextBox;
