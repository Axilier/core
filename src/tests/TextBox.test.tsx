import { mount } from 'enzyme';
import React from 'react';
import TextBox from '../components/TextBox';
import { Google } from '../components/icons';

const onChange = jest.fn();
const textBox = mount(<TextBox variant={'outlined'} />);

describe('when I render the TextBox in an app', () => {
    it('should render correctly', () => {
        expect(textBox).toMatchSnapshot();
    });
    describe('when the `label` prop is not set', () => {
        it('should not render a p element containing the label', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().className,
            ).not.toEqual('textBoxLabel');
        });
    });
    describe('when the `label` prop is set', () => {
        beforeAll(() => textBox.setProps({ label: 'test' }));
        it('should render a p element with the className textBoxLabel as the first element', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().className,
            ).toEqual('textBoxLabel');
        });
        afterAll(() => textBox.setProps({ label: undefined }));
    });
    describe('when the component is not `disabled`', () => {
        it('should have text color that #000000 either default or set by the user', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('color', '#000000');
        });
    });
    describe('when the component is `disabled`', () => {
        beforeAll(() => textBox.setProps({ disabled: true }));
        it('should have text color that #000000 either default or set by the user', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('color', '#8C8C8C');
        });
        afterAll(() => textBox.setProps({ disabled: false }));
    });
    describe('when the `variant` is outlined', () => {
        it('should have a borderRadius of 5px', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('borderRadius', '5px');
        });
        it('should have a set border', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('border', 'solid 2px #057EFF');
        });
    });
    describe('when the variant is filled', () => {
        beforeAll(() => textBox.setProps({ variant: 'filled' }));
        it('should have a borderRadius of 0', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('borderRadius', '0');
        });
        it('should have a set border', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('border', 'none');
        });
        afterAll(() => textBox.setProps({ variant: 'outlined' }));
    });
    describe('when the input is marked as `required`', () => {
        beforeAll(() => textBox.setProps({ required: true, label: 'test' }));
        it('should have a component that will render an *', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).childAt(1).props()
                    .className,
            ).toContain('requiredIcon');
        });
        afterAll(() => textBox.setProps({ required: false, label: undefined }));
    });
    describe('when the `size` is small', () => {
        beforeAll(() => textBox.setProps({ size: 'small' }));
        it('should have a set width and height', () => {
            expect(textBox.childAt(0).props().style).toHaveProperty(
                'width',
                '102px',
            );
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('height', '34px');
        });
        afterAll(() => textBox.setProps({ size: undefined }));
    });
    describe('when the `size` is large', () => {
        it('should have a set width and height', () => {
            expect(textBox.childAt(0).props().style).toHaveProperty(
                'width',
                '320px',
            );
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('height', '44px');
        });
    });
    describe('when the `height` or `width` is defined', () => {
        beforeAll(() =>
            textBox.setProps({
                size: 'small',
                height: 200,
                width: 300,
            }),
        );
        it('should overwrite the styles set by the size prop', () => {
            expect(textBox.childAt(0).props().style).toHaveProperty(
                'width',
                '300px',
            );
            expect(
                textBox.childAt(0).childAt(0).childAt(0).props().style,
            ).toHaveProperty('height', '200px');
        });
        afterAll(() =>
            textBox.setProps({
                size: undefined,
                height: undefined,
                width: undefined,
            }),
        );
    });
    describe('when the `prefixComponent` is defined', () => {
        beforeAll(() => textBox.setProps({ prefixComponent: <Google /> }));
        describe('when the `size` is also set to small', () => {
            beforeAll(() => textBox.setProps({ size: 'small' }));
            it('should set the prefixes height and width ', () => {
                expect(
                    textBox
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .props().style,
                ).toEqual({
                    marginLeft: '7px',
                });
            });
            afterAll(() => textBox.setProps({ size: 'large' }));
        });
        describe('when the `size` is also set to large', () => {
            beforeAll(() => textBox.setProps({ size: 'large' }));
            it('should set the prefixes height and width ', () => {
                expect(
                    textBox
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .props().style,
                ).toEqual({
                    marginLeft: '7px',
                });
            });
            afterAll(() => textBox.setProps({ size: undefined }));
        });
        afterAll(() => textBox.setProps({ prefixComponent: undefined }));
    });
    describe('when the `suffixComponent` is defined', () => {
        beforeAll(() => textBox.setProps({ suffixComponent: <Google /> }));
        it('should render an extra component which is the suffix component', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).childAt(0).children()
                    .length,
            ).toEqual(2);
            expect(
                textBox
                    .childAt(0)
                    .childAt(0)
                    .childAt(0)
                    .childAt(0)
                    .childAt(1)
                    .hasClass('suffix-comp'),
            ).toBeTruthy();
        });
        afterAll(() => textBox.setProps({ suffixComponent: undefined }));
    });
    describe('when the `units` is defined', () => {
        beforeAll(() => textBox.setProps({ units: 'Inches' }));
        it('should render an extra component which is the suffix component', () => {
            expect(
                textBox.childAt(0).childAt(0).childAt(0).childAt(0).children()
                    .length,
            ).toEqual(2);
            expect(
                textBox
                    .childAt(0)
                    .childAt(0)
                    .childAt(0)
                    .childAt(0)
                    .childAt(1)
                    .hasClass('unit-comp'),
            ).toBeTruthy();
        });
        describe('when the `variant` is not set', () => {
            it('should have a backgroundColor of transparent', () => {
                expect(
                    textBox
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(1)
                        .props().style,
                ).toHaveProperty('backgroundColor', 'transparent');
            });
        });
        describe('when the `variant` is set to filled', () => {
            beforeAll(() => textBox.setProps({ variant: 'filled' }));
            it('should have a backgroundColor of #F3F3F3', () => {
                expect(
                    textBox
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(0)
                        .childAt(1)
                        .props().style,
                ).toHaveProperty('backgroundColor', '#F3F3F3');
            });
            afterAll(() => textBox.setProps({ variant: undefined }));
        });
        afterAll(() => textBox.setProps({ units: undefined }));
    });
    describe('when the `onChange` prop is defined', () => {
        beforeAll(() => textBox.setProps({ onChange }));
        describe('when I type in the input', () => {
            it('should run the `onChange` function', () => {
                textBox
                    .find('input')
                    .simulate('change', { target: { value: 'Hello tester' } });
                expect(onChange).toHaveBeenCalledWith('Hello tester');
                onChange.mockClear();
            });
        });
        describe('when the filter is set for only numbers', () => {
            beforeAll(() =>
                textBox.setProps({
                    filter: (value: string) => /^\d+$/.test(value),
                }),
            );
            it('should only run onChange when the filter condition is met', () => {
                textBox
                    .find('input')
                    .simulate('change', { target: { value: 'test' } });
                expect(onChange).not.toHaveBeenCalled();
                textBox
                    .find('input')
                    .simulate('change', { target: { value: 10 } });
                expect(onChange).toHaveBeenCalledWith(10);
            });
            afterAll(() => textBox.setProps({ filter: undefined }));
        });
    });
    describe('when the `onChange` prop is not defined', () => {
        beforeAll(() => textBox.setProps({ onChange: null }));
        it('should do nothing when I type', () => {
            textBox
                .find('input')
                .simulate('change', { target: { value: 'Hello tester' } });
            expect(true).toBeTruthy();
        });
    });
});
