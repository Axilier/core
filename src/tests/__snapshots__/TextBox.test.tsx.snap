// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the TextBox in an app should render correctly 1`] = `
<TextBox
  variant="outlined"
>
  <Layout
    className="textBoxContainer"
    orientation="column"
    style={
      Object {
        "width": "320px",
      }
    }
  >
    <div
      className="textBoxContainer"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "width": "320px",
        }
      }
    >
      <Layout
        className="textBox"
        orientation="row"
        style={
          Object {
            "border": "solid 2px #057EFF",
            "borderRadius": "5px",
            "color": "#000000",
            "height": "44px",
          }
        }
      >
        <div
          className="textBox"
          style={
            Object {
              "border": "solid 2px #057EFF",
              "borderRadius": "5px",
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "height": "44px",
            }
          }
        >
          <input
            className="textBoxInput"
            maxLength={-1}
            onChange={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "text",
                "height": "44px",
                "lineHeight": "44px",
              }
            }
            value=""
          />
        </div>
      </Layout>
    </div>
  </Layout>
</TextBox>
`;
